App/controllers/user.py

```diff
@@ -1,8 +1,11 @@
-from flask import render_template
+from flask import redirect, render_template, request, url_for, jsonify
```

Comment:
controllers should never redirect, render_template, use request, URL or jsonify as these are operations that should remain on the view

Reply:
I think I have fixed this now, I have tested it and it works

CONTROLLER
# REGISTER A NORMAL USER
def register(fname,lname,email,password):
    publicUser = User(first_name=fname, last_name=lname, email=email, password=password)
    publicUser.set_password(password)
    db.session.add(publicUser)
    db.session.commit()

VIEW
# REGISTRATION NORMAL USER
@user_views.route('/register', methods=['GET','POST'])
def get_user_reg_page():
    if request.method == 'POST':
        try:
            fname = request.form['first_name']
            lname = request.form['last_name']
            email = request.form['email']
            password = request.form['password']
            RegisterUser = register(fname,lname,email,password)
        except IntegrityError:
            db.session.rollback()
            return jsonify('Something went wrong. User NOT Registered')
        return jsonify('User Registered')
        
    return render_template('register.html')